Documentation: slicing pipeline in kedro-viz

Huongg opened the pull request:
## Description

Fixes #2039. To include documentation part for slicing pipeline in kedro-voz


## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

tynandebold commented on docs/source/slice_a_pipeline.md:
```diff
@@ -0,0 +1,32 @@
+
+Slicing a pipeline refers to creating a subset of a pipeline's nodes, which can help in focusing on specific parts of the pipeline. There are two primary ways to achieve this in Kedro:
+
+1. **Programmatically with the Kedro CLI.** This method is ideal for those comfortable with command-line tools. Detailed steps on how to achieve this are available in the kedro documentation: [Slice a Pipeline](https://docs.kedro.org/en/0.19.5/nodes_and_pipelines/slice_a_pipeline.html).
```
```suggestion
1. **Programmatically with the Kedro CLI.** This method is ideal for those comfortable with command-line tools. Detailed steps on how to achieve this are available in the Kedro documentation: [Slice a Pipeline](https://docs.kedro.org/en/latest/nodes_and_pipelines/slice_a_pipeline.html).
```

Should this point to `latest` in the docs? I don't remember what the normal practice is.

astrojuanlu replied:
I don't think we're being very consistent, but at least if we want to avoid hardcoding a specific version, we should use `/stable` instead of `/latest`

tynandebold reviewed (approved):
Amazing!

One small suggestion :)

astrojuanlu commented on docs/source/slice_a_pipeline.md:
```diff
@@ -0,0 +1,32 @@
+   - Click on the first node you want to include.
+   - Hold the Shift key and select the second node.
+   
+![](./images/slice_pipeline_full_view.gif)
```
The GIF might need re-recording in light of https://github.com/kedro-org/kedro-viz/pull/2036#issuecomment-2302327774

(Also can we make it a bit bigger?)

Huongg replied:
hey yeah, I'll update the gif once I push my changes to address your comment 👍

astrojuanlu commented on docs/source/slice_a_pipeline.md:
```diff
@@ -0,0 +1,32 @@
+
+3. **View the run command:** After selecting the nodes, Kedro-Viz generates a CLI command for the sliced pipeline. You can copy this command and use it directly in your Kedro project to run the slice.
+
+4. **Slice the pipeline:** When you're ready, click the "Slice" button. This opens a new view where you can directly interact with the sliced pipeline.
```
Maybe we can have another GIF with this action + reset?

astrojuanlu reviewed (commented):
The prose looks perfect to me! There's one major comment and a suggestion from my side

rashidakanchwala commented:
Think we will need to update this doc with the newer flow.

Huongg commented:
hey @rashidakanchwala @astrojuanlu I've updated the doc with the latest flow, and removed the old gif. Let me know if you're happy with the changes 😄

astrojuanlu commented:
Both GIFs appear as broken to me https://kedro--2043.org.readthedocs.build/projects/kedro-viz/en/2043/slice_a_pipeline.html 🤔

astrojuanlu reviewed (approved):
Looks good to me, thanks a lot @Huongg !

rashidakanchwala commented on docs/source/slice_a_pipeline.md:
```diff
@@ -0,0 +1,33 @@
+# Slice a pipeline
+
+Slicing a pipeline refers to creating a subset of a pipeline's nodes, which can help in focusing on specific parts of the pipeline. There are two primary ways to achieve this in Kedro:
```
```suggestion
Slicing a pipeline in Kedro refers to creating a subset of a pipeline's nodes, which can help in focusing on specific parts of the pipeline. There are two primary ways to achieve this:
```

rashidakanchwala commented on docs/source/slice_a_pipeline.md:
```diff
@@ -0,0 +1,33 @@
+
+Slicing a pipeline refers to creating a subset of a pipeline's nodes, which can help in focusing on specific parts of the pipeline. There are two primary ways to achieve this in Kedro:
+
+1. **Programmatically with the Kedro CLI.** This method is ideal for those comfortable with command-line tools. Detailed steps on how to achieve this are available in the kedro documentation: [Slice a Pipeline](https://docs.kedro.org/en/stable/nodes_and_pipelines/slice_a_pipeline.html).
```
```suggestion
1. **Programmatically with the Kedro CLI.** This method is suitable for those comfortable with command-line tools. Detailed steps on how to achieve this are available in the kedro documentation: [Slice a Pipeline](https://docs.kedro.org/en/stable/nodes_and_pipelines/slice_a_pipeline.html).
```

rashidakanchwala commented on docs/source/slice_a_pipeline.md:
```diff
@@ -0,0 +1,33 @@
+
+1. **Programmatically with the Kedro CLI.** This method is ideal for those comfortable with command-line tools. Detailed steps on how to achieve this are available in the kedro documentation: [Slice a Pipeline](https://docs.kedro.org/en/stable/nodes_and_pipelines/slice_a_pipeline.html).
+
+2. **Visually through Kedro-Viz:** This approach allows you to visually choose and slice pipeline nodes, which then generates a run command for executing the slice within your Kedro project.
```
```suggestion
2. **Visually through Kedro-Viz:** This approach allows you to visually select and slice pipeline nodes, which then generates a run command for executing the slice within your Kedro project.
```

rashidakanchwala commented on docs/source/slice_a_pipeline.md:
```diff
@@ -0,0 +1,33 @@
+
+Kedro-Viz offers a user-friendly visual interface for slicing pipelines. Follow these steps to use the slicing feature:
+
+1. **Select elements in the flowchart:** In Kedro-Viz, choose two elements to set the boundaries for your slice:
```
```suggestion
1. **Select elements in the flowchart:** In Kedro-Viz, select two elements to set the boundaries for your slice:
```

rashidakanchwala reviewed (approved):
Looks great, just a few nits from me. 
We should add a gif and a brief description and maybe benefits on Kedro docs as well. Here - https://docs.kedro.org/en/stable/nodes_and_pipelines/slice_a_pipeline.html

thanks @Huongg

Huongg commented:
thanks @rashidakanchwala and @astrojuanlu, I've added a few lines in kedro doc to mention this feature in kedro-viz https://github.com/kedro-org/kedro/pull/4144
